datafusion/proto/src/logical_plan/mod.rs

```diff
@@ -458,23 +458,24 @@ impl AsLogicalPlan for LogicalPlanNode {
                     .map(ListingTableUrl::parse)
                     .collect::<Result<Vec<_>, _>>()?;
 
+                let partition_columns = scan
+                    .table_partition_cols
+                    .iter()
+                    .map(|col| {
+                        let arrow_type =
+                            col.arrow_type.as_ref().unwrap().try_into().map_err(|e| {
+                                proto_error(format!(
+                                    "Received an unknown ArrowType: {}",
+                                    e
+                                ))
+                            })?;
```

Comment:
I think it would be good to avoid an unwrap here as that will panic rather than returning an error on invalid protobuf

```suggestion
                        let Some(arrow_type) = col.arrow_type.as_ref() else {
                            return Err(proto_error(format!(
                                "Missing Arrow type in partition columns"
                            )));
                        };
                       let arrow_type = DataType::try_from(arrow_type).map_err(|e| {
                            proto_error(format!("Received an unknown ArrowType: {}", e))
                        })?;
```